src/classes/worker.ts

```
@@ -364,30 +364,24 @@ export class Worker<T = any> extends QueueBase {
     this.waiting && reconnect && (await this.blockingConnection.reconnect());
   }
 
-  async close(force = false) {
-    if (!this.closing) {
+  close(force = false) {
+    if (this.closing) {
+      return this.closing;
+    }
+    this.closing = (async () => {
       this.emit('closing', 'closing queue');
-      this.closing = new Promise(async (resolve, reject) => {
-        try {
-          const client = await this.blockingConnection.client;
 
-          await this.resume();
-
-          if (!force) {
-            await this.whenCurrentJobsFinished(false);
-          } else {
-            await client.disconnect();
-          }
-        } catch (err) {
-          reject(err);
-        } finally {
-          this.timerManager.clearAllTimers();
-          this.childPool && this.childPool.clean();
-        }
-        this.emit('closed');
-        resolve();
-      });
-    }
+      const client = await this.blockingConnection.client;
+      await this.pause();
+      if (!force) {
+        // ignore most failures and continue with closing
+        await this.whenCurrentJobsFinished(false).catch(err => {});
```

I think it is better to not silently ignore these errors, could lead to difficult to debug issues in the future.